BoxContentSDK/BoxContentSDK/Models/BOXRepresentation.m

```diff
@@ -0,0 +1,77 @@
+{
+    if (self = [super init]) {
+        
+        self.JSONData = JSONResponse;
+        
+        self.type = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyRepresentation
+                                                   inDictionary:JSONResponse
+                                                hasExpectedType:[NSString class]
+                                                    nullAllowed:NO];
+        
+        NSDictionary *propertiesJSON = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyProperties
+                                                                  inDictionary:JSONResponse
+                                                               hasExpectedType:[NSDictionary class]
+                                                                   nullAllowed:NO];
+        
+        self.dimensions = [propertiesJSON objectForKey:BOXAPIObjectKeyDimensions];
+        
+        self.status = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyStatus
+                                                     inDictionary:JSONResponse
+                                                  hasExpectedType:[NSString class]
+                                                      nullAllowed:NO];
+        
+        NSDictionary *detailsJSON = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyDetails
+                                                                   inDictionary:JSONResponse
+                                                                hasExpectedType:[NSDictionary class]
+                                                                    nullAllowed:NO];
+        if ([detailsJSON count] > 0) {
+            self.details = detailsJSON;
+        }
+        
+        NSDictionary *linksJSON = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyLinks
+                                                                 inDictionary:JSONResponse
+                                                              hasExpectedType:[NSDictionary class]
+                                                                  nullAllowed:NO];
+        
+        NSDictionary *contentJSON = linksJSON[BOXAPIObjectKeyContent];
+        
+        NSString *contentURLString = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyURL
+                                                             inDictionary:contentJSON
+                                                          hasExpectedType:[NSString class]
+                                                              nullAllowed:NO];
+        if (contentURLString.length > 0) {
+            self.contentURL = [NSURL URLWithString:contentURLString];
+        }
+        
+        self.contentType = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyType
+                                                          inDictionary:contentJSON
+                                                       hasExpectedType:[NSString class]
```

Comment:
No, I think it's fine to leave it as NSString because that's the type we expect in the JSON dictionary.